File: src/scripts/list-tables.ts
import dotenv from "dotenv";
import { MongoClient } from "mongodb";

dotenv.config();

async function main() {
  const uri = process.env.MONGODB_URI;
  if (!uri) throw new Error("MONGODB_URI not set");
  const client = new MongoClient(uri);
  await client.connect();
  const db = client.db(process.env.MONGODB_DB ?? "casino_marketing_demo");
  const tables = await db
    .collection("table_state_snapshots")
    .find({}, { projection: { _id: 0, tableId: 1, tableName: 1, occupancyRate: 1, patronCount: 1 } })
    .sort({ tableId: 1 })
    .toArray();
  console.log("Total tables:", tables.length);
  console.log("\nFirst 10:");
  for (const t of tables.slice(0, 10)) {
    console.log(`  ${t.tableId} - ${t.tableName} (stored occupancy=${t.occupancyRate}, patronCount=${t.patronCount})`);
  }
  console.log("\nLast 10:");
  for (const t of tables.slice(-10)) {
    console.log(`  ${t.tableId} - ${t.tableName} (stored occupancy=${t.occupancyRate}, patronCount=${t.patronCount})`);
  }
  const t19 = tables.find(
    (t) => t.tableName === "Table 19" || String(t.tableId).endsWith("19")
  );
  console.log("\nFor 'Table 19':", t19 ?? "not found");
  await client.close();
}
main();

File: server/app/scripts/list_tables.py
"""List the first/last 10 tables in ``table_state_snapshots``.

Run with::

    uv run python -m app.scripts.list_tables
"""

from __future__ import annotations

import asyncio

from app.collections import web_collections as cols
from app.db import close_client, get_db


async def main() -> None:
    db = get_db()
    tables = await (
        db[cols.tables]
        .find(
            {},
            projection={
                "_id": 0,
                "tableId": 1,
                "tableName": 1,
                "occupancyRate": 1,
                "patronCount": 1,
            },
        )
        .sort("tableId", 1)
        .to_list(length=None)
    )
    print(f"Total tables: {len(tables)}")
    print("\nFirst 10:")
    for t in tables[:10]:
        print(
            f"  {t.get('tableId')} - {t.get('tableName')} "
            f"(stored occupancy={t.get('occupancyRate')}, patronCount={t.get('patronCount')})"
        )
    print("\nLast 10:")
    for t in tables[-10:]:
        print(
            f"  {t.get('tableId')} - {t.get('tableName')} "
            f"(stored occupancy={t.get('occupancyRate')}, patronCount={t.get('patronCount')})"
        )

    t19 = next(
        (
            t
            for t in tables
            if t.get("tableName") == "Table 19" or str(t.get("tableId", "")).endswith("19")
        ),
        None,
    )
    print(f"\nFor 'Table 19': {t19 if t19 is not None else 'not found'}")


async def _entrypoint() -> None:
    try:
        await main()
    finally:
        await close_client()


if __name__ == "__main__":
    asyncio.run(_entrypoint())
